auth-api/src/auth_api/services/task.py

```diff
@@ -122,17 +123,30 @@ def _update_relationship(self, origin_url: str = None):
             # Update Org relationship
             org_id = task_model.relationship_id
             if not is_hold:
+                is_govn_affidavit_review = task_model.type == TaskTypePrefix.GOVN_REVIEW.value and \
+                                         task_model.action == TaskAction.AFFIDAVIT_REVIEW.value
                 self._update_org(is_approved=is_approved, org_id=org_id,
-                                 origin_url=origin_url)
+                                 origin_url=origin_url, is_govn_affidavit_review=is_govn_affidavit_review)
             else:
+                # Task with GOVN type and ACCOUNT_REVIEW action cannot be put on hold
+                if task_model.type == TaskTypePrefix.GOVN_REVIEW.value and \
+                        task_model.action == TaskAction.ACCOUNT_REVIEW.value:
+                    raise BusinessException(Error.INVALID_INPUT, None)
+
                 # no updates on org yet.put the task on hold and send mail to user
                 org: OrgModel = OrgModel.find_by_org_id(org_id)
                 if org.status_code != OrgStatus.PENDING_STAFF_REVIEW.value:
                     org.status_code = OrgStatus.PENDING_STAFF_REVIEW.value
                     org.flush()
+
+                # BCEID related tasks can be put on hold
                 is_bceid = org.access_type in (AccessType.EXTRA_PROVINCIAL.value, AccessType.REGULAR_BCEID.value)
-                # bceid holds need notifications
-                if is_bceid:
+                # Task with GOVN type and ACCOUNT_REVIEW action can also be put on hold
+                is_govn_affidavit_review = (org.access_type == AccessType.GOVN.value and
```

Comment:
This logic can be simple, rather than checking the ors type and all, you can drive logic based on task model values.  E.g, here it could be simple to check the task action and task status.